### PR TITLE
Refactor Blocktx PeerHandler

### PR DESCRIPTION
**Pull request by kuba-4chain**
## Description of Changes

This PR serves a purpose of decoupling PeerHandler logic from block processing logic in Blocktx, as well as to lay foundation for Reorg Support feature.

#### What has changed
1. Block processing, as well as transaction registering and requesting logic was moved to a new file called `processor.go`, similarly to the structure of metamorph.
2. PeerHandler is now responsible only for receiving message and is to be used as an interface to pass to PeerManager.
3. PeerHandler sends incoming blocks to `blockProcessCh` buffered channel, which is being read by `processor`.
4. `workerCh` was renamed to `blockRequestCh`. Logic of that channel remained unchanged.
5. Ticker resets were added to batching and interval methods.
6. Tests were fixed and improved with the new processor logic.

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [ ] I have added new unit tests
- [x] All tests pass locally
- [x] I have tested manually in my local environment

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: internal/blocktx/processor.go [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+	b := make([]byte, 8)
+
+	for i := 0; i < length; i++ {
+		b[i] = script[i+1]
+	}
+
+	return binary.LittleEndian.Uint64(b)
+}
+
+func (ph *Processor) Shutdown() {
```
Comment (arkadiuszos4chain) [inline internal/blocktx/processor.go:677]: Please move above test functions

**Review by boecklim — APPROVED** [review]
Very nice!

File: internal/blocktx/peer_handler.go [hunk 2]
```diff
@@ -89,353 +64,17 @@ func init() {
...
-	}
-
-	if len(updatedTxs) > 0 {
-		err = ph.publishMinedTxs(updatedTxs)
-		if err != nil {
-			ph.logger.Error("failed to publish mined txs", slog.String("err", err.Error()))
-		}
+func NewPeerHandler(logger *slog.Logger, blockRequestCh chan BlockRequest, blockProcessCh chan *p2p.BlockMessage) *PeerHandler {
+	return &PeerHandler{
+		logger:         logger,
```
[the remark this answers is not in the document]
Comment (kuba-4chain) [reply internal/blocktx/peer_handler.go]: Done

File: internal/blocktx/processor.go [hunk 3]
```diff
@@ -0,0 +1,685 @@
...
+	opts ...func(*Processor),
+) (*Processor, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+
+	ph := &Processor{
+		store:                       storeI,
+		logger:                      logger,
```
[the remark this answers is not in the document]
Comment (kuba-4chain) [reply internal/blocktx/processor.go]: Done

**Review by arkadiuszos4chain — APPROVED** [review]

**Pull request merged**